### PR TITLE
upload-cli: allow creation of dataset

### LINKED ISSUE #299: Allow creation of dataset during upload
At the moment a user of the Lightly Platform would need to go to the webapp and create a new dataset to get the `dataset_id` for uploading. However, it would be great if the user could just use the token and during the upload process, a new dataset would automatically be created.

### LINKED ISSUE #313: Codecoverage fluctuates due to version checking failing only sometimes
## Example problem:
https://github.com/lightly-ai/lightly/pull/312 shows a change in the code coverage
image
Same for https://github.com/lightly-ai/lightly/pull/303:
image

Both PRs did not do anything which should change code coverage, thus this should not happen. Other PRs, on the other hand, have the opposite effect, e.g. https://github.com/lightly-ai/lightly/pull/308
image

## Solution:
https://github.com/lightly-ai/lightly/blob/bd982c9513effd75c11d9b2671de09a4be6ce155/.github/workflows/test.yml#L30-L32
change it, perhaps `export LIGHTLY_SERVER_LOCATION="localhost:-1"`works.

### PR DESCRIPTION
**Pull request by MalteEbner**
closes #299 
closes #313

## Description
- changed one line in GitHub action to solve #313
- added parameter `new_dataset_name` to the `config.yaml`
- allow to use `lightly-upload` and `lightly-magic` either with creating a new dataset in the webapp (if `new_dataset_name` is defined) or uploads to an existing dataset (if `dataset_id` is defined)
- wrote test for the `upload_cli` and `lightly_cli` increasing coverage quite a lot
- added docstrings to the `upload_cli` including the new parameter

## Tests (also look at the code coverage changes)
- [x] My change is (partly) covered by existing tests
- [x] My change needs new tests 
- [x] I have added/adapted tests accordingly.
- [x] I have manually tested the change. 

Manual test procedure:
```bash
pip uninstall lightly
export BRANCH_NAME="299-upload-cli-creation-of-dataset-me"
pip install "git+https://github.com/lightly-ai/lightly.git@$BRANCH_NAME"
lightly-upload input_dir=data/ token='123' new_dataset_name='new_dataset_name_xyz'
lightly-magic input_dir=data/ token='123' new_dataset_name='new_dataset_name_xyz'
```

## Documentation
- [x] I have added docstrings to all changed/added public functions/methods.
- [x] My change requires a change to the documentation (https://docs.lighly.ai).
- [x] I have updated the documentation accordingly.
- [x] The autodocs update the documentation accordingly.

### CODE REVIEW

File: lightly/cli/config/config.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ checkpoint: ''                # Path to a model checkpoint. If left empty, a pr
 # Lightly platform.
 token: ''                     # User access token to the platform.
 dataset_id: ''                # Identifier of the dataset on the platform
+new_dataset_name: ''          # Name of the new dataset to be created on the platform
```
Comment (IgorSusmelj) [inline lightly/cli/config/config.yaml:15]: Would change to: ...created on the **Lightly** platform

**Comment by IgorSusmelj** [comment]
Looks good to me. Could you please update the CLI docs or create a new issue for this?
https://github.com/lightly-ai/lightly/blob/develop/docs/source/getting_started/command_line_tool.rst

File: docs/source/getting_started/command_line_tool.rst [hunk 2]
```diff
@@ -120,6 +125,7 @@ Upload embeddings using the CLI
 ----------------------------------
 
 You can upload embeddings directly to the Lightly Platform using the CLI.
+Again, you can use the *dataset_id* and *new_dataset_name* interchageably.
```
Comment (IgorSusmelj) [inline docs/source/getting_started/command_line_tool.rst:128]: typo intercha**n**geably

**Review by IgorSusmelj — APPROVED** [review]
small typo. Otherwise fine for me

File: docs/source/getting_started/command_line_tool.rst (hunk 2 above)
Comment (MalteEbner) [reply docs/source/getting_started/command_line_tool.rst]: solved

**Pull request merged**